src/all/pixiv/src/eu/kanade/tachiyomi/extension/all/pixiv/Pixiv.kt

```diff
@@ -0,0 +1,146 @@
+        .add("Referer", "$baseUrl/")
+
+    private val pixiv = PixivClient(if (lang == "all") "ja" else lang, client)
+    private val postCache = LruCache<Int, PixivArtwork>(50)
+
+    private fun fetchArtwork(id: Int): Observable<PixivArtwork> =
+        Observable.fromCallable { postCache.get(id) }
+            .filter { it != null }
+            .switchIfEmpty(
+                Observable.defer {
+                    pixiv.fetchArtwork(id).doOnNext { postCache.put(id, it) }
+                },
+            )
+
+    private fun fetchArtwork(url: String): Observable<PixivArtwork> =
+        fetchArtwork(getIllustIdFromUrl(url))
+
+    override fun fetchPopularManga(page: Int): Observable<MangasPage> =
+        pixiv.search("漫画", page).map { results ->
+            val mangas = results.popular
+                .run { recent.asSequence() + permanent.asSequence() }
+                .map { it.toSManga() }
+                .toList()
+
+            MangasPage(mangas, hasNextPage = false)
+        }
+
+    override fun fetchLatestUpdates(page: Int): Observable<MangasPage> =
+        fetchSearchManga(page, "漫画", filters = FilterList())
+
+    override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> {
+        var includeNonManga = false
+        var ascendingSort = false
+
+        filters.forEach {
+            when (it) {
+                is NonMangaFilter -> includeNonManga = it.state
+                is SortFilter -> ascendingSort = it.state!!.ascending
+                else -> throw IllegalStateException("Filter unrecognized")
+            }
+        }
+
+        val type = if (includeNonManga) "artworks" else "manga"
+
+        return pixiv.search(query, page, type, ascendingSort).map { results ->
+            val mangas = results.artworks.data
+                .asSequence()
+                .filterNot { it.isAdContainer }
+                .map { it.toSManga() }
+                .toList()
+
+            MangasPage(mangas, hasNextPage = mangas.isNotEmpty())
+        }
+    }
+
+    override fun fetchMangaDetails(manga: SManga): Observable<SManga> =
+        fetchArtwork(manga.url).map { it.toSManga() }
+
+    override fun fetchChapterList(manga: SManga): Observable<List<SChapter>> =
+        fetchArtwork(manga.url).map { artwork ->
+            listOf(
+                SChapter.create().apply {
+                    url = manga.url
+                    name = "Oneshot"
+                    date_upload = parseTimestamp(artwork.uploadDate)
+                    chapter_number = 0F
+                },
+            )
+        }
+
+    override fun fetchPageList(chapter: SChapter): Observable<List<Page>> =
+        pixiv.fetchPages(getIllustIdFromUrl(chapter.url)).map { pages ->
+            pages.mapIndexed { i, page ->
+                Page(index = i, imageUrl = page.urls.original)
+            }
+        }
+
+    override fun getChapterUrl(chapter: SChapter): String = chapter.url
```

Comment:
Should add the `getMangaUrl` as well, as by default it calls `mangaDetailsRequest` and it will throw an exception. I also included the `baseUrl` in the methods as a follow-up to the first change request.

```suggestion
    override fun getMangaUrl(manga: SManga): String = baseUrl + manga.url

    override fun getChapterUrl(chapter: SChapter): String = baseUrl + chapter.url
```